internal/cli/cmd/dac/setup/setup.go

```diff
@@ -0,0 +1,249 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package setup
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+
+	persesCMD "github.com/perses/perses/internal/cli/cmd"
+	"github.com/perses/perses/internal/cli/config"
+	"github.com/perses/perses/internal/cli/output"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"golang.org/x/mod/semver"
+)
+
+const archiveName = "sources.tar.gz"
+const depsFolderName = "cue/"
+const depsRootDstPath = "cue.mod/pkg/github.com/perses/perses" // for more info see https://cuelang.org/docs/concepts/packages/
+const maxFileSize = 10240                                      // = 10 KiB. Estimated max size for CUE files. Limit required by gosec.
+
+type option struct {
+	persesCMD.Option
+	writer  io.Writer
+	version string
+}
+
+func (o *option) Complete(args []string) error {
+	if len(args) > 0 {
+		return fmt.Errorf("no args are supported by the command 'setup'")
+	}
+
+	// If no version provided, it should default to the version of the Perses server
+	if o.version == "" {
+		logrus.Debug("version flag not provided, retrieving version from Perses server..")
+		apiClient, err := config.Global.GetAPIClient()
+		if err != nil {
+			return fmt.Errorf("you need to either provide a version or be connected to a Perses server")
+		}
+
+		health, err := apiClient.V1().Health().Check()
+		if err != nil {
+			logrus.WithError(err).Debug("can't reach Perses server")
+			return fmt.Errorf("can't retrieve version from Perses server")
+		}
+		o.version = health.Version
+	}
+
+	// Validate the format of the provided version
+	if !semver.IsValid(fmt.Sprintf("v%s", o.version)) {
```

Comment:
I would put this step in the `Validate` function. It think it fits nicely there

Reply:
Also perhaps we should check if the version start already with a `v`. You could have both syntax supported `v0.42.0` and `0.42.0`